### DISCUSSION TITLE
Custom CSV and Markdown Loader Breaks on Reload

### DISCUSSION THREAD

**Discussion by an0nusr** [webpack/webpack#17593] [Q&A]
I have a very simple webpack loader that transforms a csv file (that has already been parsed by csv-loader) and processes any markdown in the csv's description field:

```js
// markdown-loader.js
var md = require("markdown-it")()

module.exports = function (csvData) {
    let newCSVdata = eval(csvData)
        .map(x => ({...x, Description: md.renderInline(x.Description)}))
    return `export default ${JSON.stringify(newCSVdata)}`
}
```

When I start the project with `webpack serve` the resource is properly built and everything works properly. However, if I make changes to the CSV and save them, webpack immediately fails and I get:

```
ERROR in ./data.csv
Module build failed (from ./markdown-loader.js):
LoaderRunnerError: Module '[PATH TO markdown-loader.js]' is not a loader (must have normal or pitch function)
```

As far as I can tell, the module has a function, and it works normally when making a production build, or on first build. It's just on reload operations (whenever the CSV is changed) that it breaks. 

Here is the webpack config:
```js
const path = require('path');

module.exports = {
  entry: './src/index.js',
  output: {
    filename: 'main.js',
    path: path.resolve(__dirname, 'dist'),
  },
  mode: "production",

  devServer: {
    static: {
      directory: path.join(__dirname, 'dist'),
    },
    compress: true,
    port: 9000,
  },

  module: {
    rules: [
      {
        test: /\.csv$/,
        use: [
        { loader: path.resolve("./markdown-loader.js") },
        {
          loader: 'csv-loader',
          options: {
            dynamicTyping: true,
            header: true,
            skipEmptyLines: true
          }
        }
        ]
      }
    ]
  }
};
```

Is there something I'm missing?

**Comment by alexander-akait** [comment]
Sorry, still valid?

**Reply by an0nusr** [reply]
Yes, this issue appears to be happening with webpack: 5.89.0. 

My solution was to rewrite the loader so instead of chaining a CSV loader and the markdown loader, there was only one that did both. That fixed the problem and the page updates if I update the CSV. 

However, I am still curious why this problem occurred. The [webpack docs seem to encourage chaining small loaders](https://webpack.js.org/contribute/writing-a-loader/) rather than doing a multiple steps in one, so it seems like my fix is bad practice...

For reference, here is the updated, dual-purpose working loader.

```js
var md = require("markdown-it")({
    breaks: false //set to true to enable line breaks in descriptions (otherwise linebreaks won't be converted to <br>)
})

var papa = require("papaparse")

module.exports = function (works) {
    // consume the csv
    let parsedCSV = papa.parse(works, {
        header: true,
        skipEmptyLines: true
    })

    //console.log(parsedCSV.data)

    let newworks = parsedCSV.data
        .map(x => ({...x, Description: md.renderInline(x.Description)})) //parse markdown in description
    
    //console.log(newworks)
    return `export default ${JSON.stringify(newworks)}`
}
```
